File: examples/docs/modules/ROOT/examples/akkastreams-scala/src/main/scala/cloudflow/akkastreamsdoc/DataMerge.scala
package cloudflow.akkastreamsdoc

import cloudflow.akkastream._
import cloudflow.streamlets.avro._
import cloudflow.streamlets._

import cloudflow.akkastream.util.scaladsl.MergeLogic

class DataMerge extends AkkaStreamlet {

  val in0 = AvroInlet[Data]("in-0")
  val in1 = AvroInlet[Data]("in-1")
  val out = AvroOutlet[Data]("out", d ⇒ d.key)

  final override val shape = StreamletShape.withInlets(in0, in1).withOutlets(out)

  override final def createLogic = new MergeLogic(Vector(in0, in1), out)
}

File: examples/docs/modules/ROOT/examples/akkastreams-java/src/main/java/cloudflow/akkastreamsdoc/DataMerge.java
package cloudflow.akkastreamsdoc;

// end::merge[]
import cloudflow.akkastream.*;
import cloudflow.streamlets.*;
import cloudflow.streamlets.avro.*;
import cloudflow.akkastream.javadsl.*;
import cloudflow.akkastream.util.javadsl.MergeLogic;

import java.util.*;

class DataMerge extends AkkaStreamlet {
  AvroInlet<Data> inlet1 = AvroInlet.<Data>create("in-0", Data.class);
  AvroInlet<Data> inlet2 = AvroInlet.<Data>create("in-1", Data.class);
  AvroOutlet<Data> outlet = AvroOutlet.<Data>create("out",  d -> d.getKey(), Data.class);

  public StreamletShape shape() {
    return StreamletShape.createWithInlets(inlet1, inlet2).withOutlets(outlet);
  }
  public MergeLogic createLogic() {
    List<CodecInlet<Data>> inlets = new ArrayList<CodecInlet<Data>>();
    inlets.add(inlet1);
    inlets.add(inlet2);
    return new MergeLogic(inlets, outlet, getContext());
  }
}
// end::merge[]